flocker/apiclient/_client.py

```diff
@@ -744,6 +779,21 @@ def list_containers_configuration(self):
         )
         return d
 
+    def list_containers_state(self):
+        d = self._request(b"GET", b"/state/containers", None, {OK})
+
+        def parse(container):
+            return ContainerState(
+                node_uuid=UUID(container['node_uuid']),
```

Comment:
I have added a more specific error if the server returns a malformed response.  It may only be one call, but we can fix it in other places as we modify them.